analyze/timestamp.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+
+import config
+import io
+from pathlib import Path
+import subprocess
+import sys
+import yaml
+
+from enum import Enum
+
+__all__ = ["Timestamp", "Subsystem", "TimestampParser"]
+
+
+class Subsystem(Enum):
+    LIBTEEC = 1
+    LINUXMOD = 2
+    OPTEECORE = 3
+
+
+class Addr2Line(object):
+    def __init__(self, binary):
+        if not (isinstance(config.addr2line, Path) and
+                config.addr2line.is_file()):
+            raise SystemError('addr2line binary was not found')
+        if not (isinstance(binary, Path) and binary.is_file()):
+            raise ValueError(str(binary) + " was not found")
+
+        self.process = subprocess.Popen([str(config.addr2line),
+                                        "-e", str(binary)],
+                                        stdin=subprocess.PIPE,
+                                        stdout=subprocess.PIPE)
+        # and lets use buffered input
+        self.stdin_wrapper = io.TextIOWrapper(self.process.stdin,
+                                              'utf-8')
+
+    def __del__(self):
+        if self.process:
+            self.process.terminate()
+
+    def lookup(self, addr):
+        try:
+            print(hex(addr))
+            self.stdin_wrapper.write(hex(addr) + '\n')
+            self.stdin_wrapper.flush()
+            info = self.process.stdout.readline().decode("utf-8")
+            print(info)
+        except IOError:
+            raise RuntimeError("Can't communicate with addr2line")
+        finally:
+            ret = self.process.poll()
+            if ret:
+                raise RuntimeError(
+                    "addr2line terminated unexpectedly")
+
+        (file, line) = info.rsplit(":", 1)
+
+        if file == "??":
+            raise RuntimeError("Illegal binary file")
+        if line == "0":
+            raise RuntimeError("Illegal line")
+
+        return (file, int(line))
+
+
+class Timestamp(object):
+    """Used to store a timestamp
+
+    Able to parse address of binary file of subsystem and provide
+    actual source file and line
+    """
+    __symbol_parsers = {}
+
+    def __init__(self, core, counter, subsystem, address, payload):
+        self.core = core
+        self.counter = counter
+        if subsystem is not Subsystem:
+            raise TypeError("Wrong subsystem type")
+
+        self.subsystem = subsystem
+        self.address = long(address)
+        self.payload = payload
+
+        self.__parse_addr()
+
+    def __parse_addr(self):
```

Comment:
Any specific reason why you are putting prefix `__foo` on non-standard Python functions in classes like this?

Reply:
@jbech-linaro 
I'm using double underscore for a name mangling , to avoid by error this "private" method invocations (as the method name will be mangled) outside the class (although PEP8 recommends to use single-leading-underscore convention instead for this purposes, and name mangling has a bit different purpose). I'll change it to single-leading-underscore, if we decided that single-leading-underscore is more preferable for such cases